### PR TITLE
update snark input

### PR DESCRIPTION
**Pull request by invocamanman**
- Update snark Input

### CODE REVIEW

File: contracts/ProofOfEfficiency.sol [hunk 1]
```diff
@@ -370,18 +375,46 @@ contract ProofOfEfficiency {
...
+        bytes memory snarkHashBytes;
+        assembly {
+            // Set snarkHashBytes to the next free memory pointer
+            snarkHashBytes := mload(0x40)
+
+            // Reserve the memory. 32 for the length , the input bytes and 32
+            // extra bytes at the end for word manipulation
+            mstore(0x40, add(add(snarkHashBytes, 0x40), _SNARK_SHA_BYTES))
+
+            // Set the actua length of the input bytes
```
Comment (krlosMata) [inline contracts/ProofOfEfficiency.sol:387]:
```suggestion
            // Set the actual length of the input bytes
```

File: contracts/ProofOfEfficiency.sol [hunk 2]
```diff
@@ -5,6 +5,7 @@ import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
 import "@openzeppelin/contracts/token/ERC20/extensions/ERC20Burnable.sol";
 import "./interfaces/IVerifierRollup.sol";
 import "./interfaces/IGlobalExitRootManager.sol";
+import "hardhat/console.sol";
```
Comment (krlosMata) [inline contracts/ProofOfEfficiency.sol:8]: I guess it was for debugging xD

File: contracts/ProofOfEfficiency.sol [hunk 3]
```diff
@@ -370,18 +375,46 @@ contract ProofOfEfficiency {
...
+            snarkHashBytes := mload(0x40)
+
+            // Reserve the memory. 32 for the length , the input bytes and 32
+            // extra bytes at the end for word manipulation
+            mstore(0x40, add(add(snarkHashBytes, 0x40), _SNARK_SHA_BYTES))
+
+            // Set the actua length of the input bytes
+            mstore(snarkHashBytes, _SNARK_SHA_BYTES)
+
+            // Set the pointer at the begining of the byte array
```
Comment (krlosMata) [inline contracts/ProofOfEfficiency.sol:390]:
```suggestion
            // Set the pointer at the beginning of the byte array
```

File: test/contracts/proofOfEfficiency.test.js [hunk 4]
```diff
@@ -713,14 +713,7 @@ describe('Proof of efficiency', () => {
...
-            newLocalExitRoot,
-            newStateRoot,
-            numBatch,
-        );
-        expect(circuitNextInputSC).to.be.equal(circuitInputSC);
-        expect(circuitNextInputSC).to.be.equal(circuitInputJS);
+        expect(circuitInpuStarktSC).to.be.equal(circuitInputStarkJS);
 
         // Check snark input
         const inputSnarkSC = await proofOfEfficiencyContract.calculateSnarkInput(
```
Comment (krlosMata) [inline test/contracts/proofOfEfficiency.test.js:719]: Add unitary test checking values compared with JS version in https://github.com/0xPolygonHermez/zkevm-commonjs/blob/main/test/contract-utils.test.js#L16

**Review by krlosMata — CHANGES_REQUESTED** [review]

**Review by krlosMata — APPROVED** [review]
:smile_cat:

**Review by ignasirv — APPROVED** [review]
💯

**Pull request merged**